spec/services/rally_spec.rb

```diff
@@ -14,9 +14,61 @@
     # stub_request(:get, //).
     #   to_return(status: 200, body: raw_fixture("rally/tbd.json"))
   end
+  let(:service_params) do
+    {}
+  end
 
-  it "can be installed"
+  let(:webhook_payload) do
+    Hashie::Mash.new JSON.parse(fixture("rally/rally_update_webhook.json").read)
+  end
+
+  let (:webhook_service) do
+    class ApiStub
+      def initialize integration_fields
+        @integration_fields = integration_fields
+      end
+
+      def search_integration_fields x, y, z
+        @integration_fields
+      end
+
+      def put x, y
+      end
+    end
+    integration_fields = Hashie::Mash.new JSON.parse(fixture("rally/aha_integration_fields.json").read)
+    api = ApiStub.new integration_fields
+    service = AhaServices::Rally.new service_params
+    service.extend(AhaServices::RallyWebhook)
+    service.instance_variable_set(:@payload, Hashie::Mash.new(webhook_payload))
+    service.instance_variable_set(:@api, api)
+    service
+  end
+
+  let (:service) do
+    AhaServices::Rally.new service_params
+  end
 
+  let(:h_req_service) do
+    RallyHierarchicalRequirementResource.new service
+  end
+
+
+
+  it "creates the right tag queries" do
+    expect(h_req_service.send( :build_tag_query, ["1", "2", "3", "4", "5", "6"] ))
+      .to eq '((((((Name = "1") OR (Name = "2")) OR (Name = "3")) OR (Name = "4")) OR (Name = "5")) OR (Name = "6"))'
+  end
+
+  it "adds tags" do
+    api = webhook_service.api
+    def api.put resource, update_hash
```

Comment:
Go ahead and use Rspec's message expectations here instead of redefining the method.

`expect(api).to receive(:put) { |_, update_hash| expect(update_hash["feature"][:tags]).to be_present}`